adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -726,6 +726,93 @@ public void onError(Exception exc) {
+     * @param forceRefresh when true, access token is renewed using broker if available; otherwise, uses local refresh token
+     * @return A {@link Future} object representing the
+     * {@link AuthenticationResult} of the call. It contains Access
+     * Token,the Access Token's expiration time, Refresh token, and
+     * {@link UserInfo}.
+     * @throws AuthenticationException If silent request fails to get the token back.
+     * @throws InterruptedException    If the main thread is interrupted before or during the activity.
+     */
+    public AuthenticationResult acquireTokenSilentSync(String resource, String clientId, String userId, boolean forceRefresh)
+            throws AuthenticationException, InterruptedException {
+
+        final String methodName = ":acquireTokenSilentSync";
+        checkPreRequirements(resource, clientId);
+        checkADFSValidationRequirements(null);
+        final AtomicReference<AuthenticationResult> authenticationResult = new AtomicReference<>();
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        final String requestId = Telemetry.registerNewRequest();
+        final APIEvent apiEvent = createApiEvent(mContext, clientId, requestId, EventStrings.ACQUIRE_TOKEN_SILENT_SYNC);
+        apiEvent.setPromptBehavior(PromptBehavior.Auto.toString());
+
+        final AuthenticationRequest request = new AuthenticationRequest(mAuthority, resource,
+                clientId, userId, getRequestCorrelationId(), getExtendedLifetimeEnabled(), forceRefresh);
+        request.setSilent(true);
+        request.setPrompt(PromptBehavior.Auto);
+        request.setUserIdentifierType(UserIdentifierType.UniqueId);
+        request.setTelemetryRequestId(requestId);
+
+        final Looper currentLooper = Looper.myLooper();
+        if (currentLooper != null && currentLooper == mContext.getMainLooper()) {
+            Logger.e(TAG + methodName,
+                    "Sync network calls must not be invoked in main thread. "
+                            + "This method will throw android.os.NetworkOnMainThreadException in next major release",
+                    new NetworkOnMainThreadException());
+        }
+        createAcquireTokenRequest(apiEvent).acquireToken(null, false, request,
+                new AuthenticationCallback<AuthenticationResult>() {
+                    @Override
+                    public void onSuccess(AuthenticationResult result) {
+                        authenticationResult.set(result);
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void onError(Exception exc) {
+                        exception.set(exc);
+                        latch.countDown();
+                    }
+                });
+
+        latch.await();
+
+        Exception e = exception.get();
+        if (e != null) {
```

Comment:
Haven't traced the entirety of this code to be sure, but I suspect you'll need to set the success state on the `apiEvent` in `onSuccess` and `onError` -- if there is an `Exception` and message, you'll want to set these values too. See `setWasApiCallSuccessful` and `stopTelemetryAndFlush`